gufe/tests/test_smallmoleculecomponent.py

```diff
@@ -182,4 +183,32 @@ def test_sorting(ethane, alt_ethane):
     order1 = [ethane, alt_ethane, ethane]
     order2 = [alt_ethane, ethane, ethane]
 
-    assert sorted(order1) == sorted(order2)
\ No newline at end of file
+    assert sorted(order1) == sorted(order2)
+
+
+class TestSmallMoleculeSerialization:
+    def test_to_dict(self, phenol):
+        d = phenol.to_dict()
+
+        assert isinstance(d, dict)
+
+    def test_deserialize_roundtrip(self, toluene, phenol):
+        # TODO: Currently roundtripping via openff adds Hydrogens even when
+        #       they weren't in the original input molecule.
+        roundtrip = SmallMoleculeComponent.from_dict(phenol.to_dict())
+
+        assert roundtrip == phenol
```

Comment:
Our hash is purely smiles, could you check coordinates too please? (just in case)